tests/drivers/random/randomnumgen_basic_api/src/main.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2017 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <random.h>
+#include <ztest.h>
+
+#define BUFFER_LENGTH 10
+
+/*
+ * Function invokes the get_entropy callback in driver
+ * to get the random data and fill to passed buffer
+ */
+static int get_entropy(void)
+{
+	struct device *dev;
+	u8_t buffer[BUFFER_LENGTH] = {0};
+	int r;
+
+	TC_PRINT("Random Example! %s\n", CONFIG_ARCH);
+	dev = device_get_binding(CONFIG_RANDOM_NAME);
+	if (!dev) {
+		TC_PRINT("error: no random device\n");
+		return TC_FAIL;
+	}
+
+	TC_PRINT("random device is %p, name is %s\n",
+			dev, dev->config->name);
+
+	/* The BUFFER_LENGTH-1 is used so the driver will not
+	 * write the last byte of the buffer. If that last
+	 * byte is not 0 on return it means the driver wrote
+	 * outside the passed buffer, and that should never
+	 * happen.
+	 */
+	r = random_get_entropy(dev, buffer, BUFFER_LENGTH-1);
+	if (r) {
+		TC_PRINT("Error: random_get_entropy failed: %d\n", r);
+		return TC_FAIL;
+	}
+	if (buffer[BUFFER_LENGTH-1] != 0) {
```

Comment:
Might be worthwhile to check if the buffer was completely filled as well, by counting number of 0 bytes.

Reply:
@lpereira , i think we should not check with buffer completely filled as it is random number so possibly any byte can come as 0. Then in this case if till 9 bytes are non-zero and last is zero then with this way test case actually passed but test shows failed.
We can do one thing that we can check whether we got any non-zero number or not if yes we actually got random number.Not sure whether this is a better way to check.As random number can give any number at any point.

Reply:
@lpereira i tried to check whether buffer was completly filled or not by checking number of 0. Few time i seen one of the byte location got 0. But i am not sure whether this location is not filled or this location is filled by 0 itself. What do you think should i check every byte with 0.
Below is the one of the output with 0 byte.
Random Example! xtensa
random device is 0x3ffb0000, name is RANDOM_0
  0x5d
  0xb5
  0xd6
  0x26
  0x00
  0x82
  0x9f
  0x08
  0x3f

Reply:
This test shouldn't test the quality of the random number device: it should be just a black box test to test the drivers.

The idea here is not really to check if the random number generator generated one or more zeros.  They're as much of a number as anything else.  The idea is to test if the drivers are capable of filling more than a few bytes; for instance, the ESP32 random device, generates random numbers in 4-byte quantities, and it's up to the driver to obtain as much 4-byte quantities as possible and write to the proper locations.

That said, we could test like so:

   * fill buffer with zero bytes
   * fill it up with random bytes
   * if less than 20% of the buffer is zero, it's good enough for this test
   * otherwise, fill it up with a byte other than zero; say, 0xa5
   * fill it up again with random bytes
   * the test passes if more than 80% of the bytes are not 0xa5

I think this is sufficient; what do you think?

Reply:
@lpereira  Last time i thought you are asking about checking with zero only. I did not think about this way.Yes this way atleast we can verify whether all bytes of buffer is completly filled or not.
Updated the changes as you mentioned above.